### PR TITLE
Refactoring related to new provider types integration

### PR DESCRIPTION
**Pull request by EugeneButusov**

### CODE REVIEW

File: SoomlaAndroidProfile/src/com/soomla/profile/SoomlaProfile.java [hunk 1]
```diff
@@ -109,12 +128,81 @@ public boolean initialize(Activity activity, boolean usingExternalProvider, Map<
...
             return false;
         }
-        mAuthController.settleAutoLogin(activity);
-        mSocialController.settleAutoLogin(activity);
+        this.settleAutoLogin(activity);
 
         return true;
     }
 
+    public void settleAutoLogin(Activity activity) {
```
Comment (borisspektor) [inline SoomlaAndroidProfile/src/com/soomla/profile/SoomlaProfile.java:136]: @EugeneButusov this should be private I believe

File: SoomlaAndroidProfile/src/com/soomla/profile/social/ISocialProvider.java [hunk 2]
```diff
@@ -19,12 +19,13 @@
...
 
 import com.soomla.profile.auth.IAuthProvider;
+import com.soomla.profile.domain.IProvider;
 
 /**
  * A provider that exposes social capabilities such as sharing, fetching user
  * feeds, uploading images etc.
  */
-public interface ISocialProvider extends IAuthProvider {
+public interface ISocialProvider extends IProvider {
```
Comment (borisspektor) [inline SoomlaAndroidProfile/src/com/soomla/profile/social/ISocialProvider.java:28]: @EugeneButusov why do we need to associate ISocialProvider with IProvider here? Didn't we want everything to be unrelated?

Comment (EugeneButusov) [inline SoomlaAndroidProfile/src/com/soomla/profile/social/ISocialProvider.java:28]: @borisspektor, it implemented as in ios-profile, you confirmed removing of interface inheritance.

File: SoomlaAndroidProfile/src/com/soomla/profile/SoomlaProfile.java (hunk 1 above)
Comment (EugeneButusov) [inline SoomlaAndroidProfile/src/com/soomla/profile/SoomlaProfile.java:136]: @borisspektor, sure, thanks.

File: SoomlaAndroidProfile/src/com/soomla/profile/social/ISocialProvider.java (hunk 2 above)
Comment (borisspektor) [inline SoomlaAndroidProfile/src/com/soomla/profile/social/ISocialProvider.java:28]: @EugeneButusov I did confirm removing. My question is why does it need to extend anything at all?

Comment (EugeneButusov) [inline SoomlaAndroidProfile/src/com/soomla/profile/social/ISocialProvider.java:28]: @borisspektor, answered you privately :)

File: SoomlaAndroidProfile/src/com/soomla/profile/SoomlaProfile.java (hunk 1 above)
Comment (EugeneButusov) [inline SoomlaAndroidProfile/src/com/soomla/profile/SoomlaProfile.java:136]: @borisspektor, pushed into GPGS branch.

**Pull request merged**